rust-toolchain

```diff
@@ -0,0 +1 @@
+1.49.0
```

Comment:
What is the purpose of this file?

Reply:
This file enforces everyone working on hashbrown to use the 1.49 Rust tool chain.

Reply:
Unfortunately this breaks our CI. Could you remove this file?

Many people, including myself, work with Rust nightly and need support for nightly features in our work. This includes working with hashbrown which uses several nightly-only features when the `"nightly"` feature is enabled.

Reply:
@Amanieu Done ✅